books/free-programming-books-langs.md

```diff
@@ -88,7 +88,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [J](#j)
 * [Java](#java)
   * [Codename One](#codename-one)
-  * [Java Reporting](#java-reporting)
+  * [Java Reporting](#javareports)
```

Comment:
why?

Reply:
Seems a bad merge with #6493, or at least start from the same point